src/providers/firestore.ts

```diff
@@ -23,6 +23,7 @@
 import { posix } from 'path';
 import * as _ from 'lodash';
 import * as firebase from 'firebase-admin';
+import * as firestore from '@google-cloud/firestore';
```

Comment:
sorry i previously missed this, but we don't actually pull in @google-cloud/firestore as a dependency ,since it's already a part of firebase-admin. So please remove this line and change line 74 to "ref: admin.firestore.DocumentReference". You'll have to test this, but I'm pretty sure that will work.

Reply:
Never mind, I take that back. It turns out firebase-admin only exposes firestore types that have a public constructor, which does not include DocumentReference. 

I think the way you have it should be fine, but I'm not sure if it's against best practice to require a module that is not in package.json, but is a sub-dependency. @inlined Can you comment on this?

Reply:
@laurenzlong You're right. I tried to do that, but `firebase-admin` doesn't expose `DocumentReference`.

IMHO, I'm not pretty sure if it's a good idea to require a module that is not in package.json. What do you think about adding `@google-cloud/firestore` as a `devDependency` explicitly? I think there is no other way to access type declaration of `DocumentReference`.

Reply:
1. This is a bug on Firestore/firebase-admin to export all of their types. We can work with @schmidt-sebastian. It seems like firebase-admin could use some of the tricks we use in firebase-functions to reexport namespaces.
2. If we don't actually `require` the module in the transpiled JavaScript I'm fine moving this as a dev dependency, though we might want to track an issue to clean this up. I suspect it might be elided since firestore is never used directly, though I can suggest some tricks if it is showing up.

Reply:
@hiranya911 regarding types in firebase-admin.

Reply:
FYI fix in firebase-admin has been merged to master: https://github.com/firebase/firebase-admin-node/pull/193f

Reply:
@laurenzlong Thanks for letting me know. I'll update this PR once the next version of `firebase-admin` is released.